[vtk] Adjust an include path to work around a dependency cycle

A file in Common/Core includes a header from Common/DataModel.
Instead of using a relative include that violates layering, we'll
include it from the current directory instead, and customize our
package.BUILD.bazel file to be sure it's available there using the
//:VTK__CommonDataModel_vtkDataObject label.

This situation is unique to Bazel, so we do not plan to upstream
any changes related to this problem.

--- Common/Core/vtkInformationDataObjectKey.cxx
+++ Common/Core/vtkInformationDataObjectKey.cxx
@@ -15,7 +15,7 @@
 #include "vtkInformationDataObjectKey.h"
 
 #if defined(vtkCommonDataModel_ENABLED)
-#include "../DataModel/vtkDataObject.h"
+#include /* ../DataModel/ */ "vtkDataObject.h"
 #endif
 
 //------------------------------------------------------------------------------
